File: classic-guild-bank/src/app/models/guildbank/bagslot.ts
import { Item } from "./item";

export class BagSlot {
    public slotId: number;
    public bagId: number;
    public quantity: number;
    public item?: Item;  
    
    constructor(init?: Partial<BagSlot>) {

        if( init ) {
            Object.assign(this, init);
        }
    }
}

File: ClassicGuildBankData/Models/BagSlot.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace ClassicGuildBankData.Models
{
    public class BagSlot
    {
        public int SlotId { get; set; }
        public int Quantity { get; set; }
        [JsonIgnore]
        public Guid BagId { get; set; }

        [NotMapped]
        [JsonProperty("bagId")]
        public string _BagId { get => BagId.ToString(); set => BagId = new Guid(value); }

        [JsonIgnore]
        public int? ItemId { get; set; }

        [JsonIgnore]
        public Bag Bag { get; set; }

        public Item Item { get; set; }
    };
}
